docs/getting-started-scalardb-analytics-postgresql.md

```diff
@@ -0,0 +1,540 @@
+              3 | Suzuki Ichiro |        10000 |         8500
+   (3 rows)
+   ```
+
+1. Read sample data in the `order.orders` table.
+
+   ```sql
+   SELECT * FROM "order".orders;
+   ```
+
+   [Command execution result]
+
+   ```sql
+   scalardb=# SELECT * FROM "order".orders;
+    customer_id |   timestamp   |               order_id
+   -------------+---------------+--------------------------------------
+              1 | 1700124015601 | 5ae2a41b-990d-4a16-9700-39355e29adf8
+              1 | 1700124021273 | f3f23d93-3862-48be-8a57-8368b7c8689e
+              2 | 1700124028182 | 696a895a-8998-4c3b-b112-4d5763bfcfd8
+              2 | 1700124036158 | 9215d63a-a9a2-4471-a990-45897f091ca5
+              3 | 1700124043744 | 9be70cd4-4f93-4753-9d89-68e250b2ac51
+              3 | 1700124051162 | 4e8ce2d2-488c-40d6-aa52-d9ecabfc68a8
+              3 | 1700124058096 | 658b6682-2819-41f2-91ee-2802a1f02857
+              3 | 1700124071240 | 4e2f94f4-53ec-4570-af98-7c648d8ed80f
+   (8 rows)
+   ```
+
+1. Read sample data in the `order.statements` table.
+
+   ```sql
+   SELECT * FROM "order".statements;
+   ```
+
+   [Command execution result]
+
+   ```sql
+   scalardb=# SELECT * FROM "order".statements;
+                  order_id               | item_id | count
+   --------------------------------------+---------+-------
+    5ae2a41b-990d-4a16-9700-39355e29adf8 |       2 |     2
+    5ae2a41b-990d-4a16-9700-39355e29adf8 |       1 |     3
+    f3f23d93-3862-48be-8a57-8368b7c8689e |       5 |     1
+    696a895a-8998-4c3b-b112-4d5763bfcfd8 |       4 |     1
+    696a895a-8998-4c3b-b112-4d5763bfcfd8 |       3 |     1
+    9215d63a-a9a2-4471-a990-45897f091ca5 |       2 |     1
+    9be70cd4-4f93-4753-9d89-68e250b2ac51 |       1 |     1
+    4e8ce2d2-488c-40d6-aa52-d9ecabfc68a8 |       2 |     1
+    658b6682-2819-41f2-91ee-2802a1f02857 |       3 |     1
+    4e2f94f4-53ec-4570-af98-7c648d8ed80f |       5 |     1
+   (10 rows)
+   ```
+
+1. Read sample data in the `order.items` table.
+
+   ```sql
+   SELECT * FROM "order".items;
+   ```
+
+   [Command execution result]
+
+   ```sql
+   scalardb=# SELECT * FROM "order".items;
+    item_id |  name  | price
+   ---------+--------+-------
+          5 | Melon  |  3000
+          2 | Orange |  2000
+          4 | Mango  |  5000
+          1 | Apple  |  1000
+          3 | Grape  |  2500
+   (5 rows)
+   ```
+
+1. Run the `JOIN` query.
+
+   For example, you can see the order history of each user as follows.
+
+   ```sql
+   SELECT
+     c.customer_id,
+     c.name,
+     o.order_id,
+     i.name,
+     s.count
+   FROM
+     customer.customers AS c
+     LEFT OUTER JOIN
+       "order".orders AS o
+       ON c.customer_id = o.customer_id
+         LEFT OUTER JOIN
+           "order".statements AS s
+           ON o.order_id = s.order_id
+             LEFT OUTER JOIN
+               "order".items AS i
+               ON s.item_id = i.item_id
+   ORDER BY c.customer_id, o.timestamp, o.order_id;
```

Comment:
Using window function might be good example. Like this? (sorry if it doesn't work in advance)

```sql
select * from (
    select c.name, c.credit_limit - c.credit_total as remaining, array_agg(i.name) over (partition by c.name) as items
    from orders o
    join customers c on o.customer_id = c.customer_id
    join statements s on o.order_id = s.order_id
    join items i on s.item_id = i.item_id
) x group by name, remaining, items
```

And I think this query can be improved...

Reply:
I suppose, to show how ScalarDB Analytics can extend the capability, the query should include complex joins, aggregations, and sorting. I'm not familiar with the dataset of this example, so I'll try it later and consider good examples.

Reply:
@komamitsu  
Thank you for your suggestion! I modified it a bit mainly to fix syntax errors on actual PostgreSQL, but I was able to run it as follows!

```sql
scalardb=# SELECT * FROM (
scalardb(#     SELECT c.name, c.credit_limit - c.credit_total AS remaining, array_agg(i.name) OVER (PARTITION BY c.name) AS items
scalardb(#     FROM "order".orders o
scalardb(#     JOIN customer.customers c ON o.customer_id = c.customer_id
scalardb(#     JOIN "order".statements s ON o.order_id = s.order_id
scalardb(#     JOIN "order".items i ON s.item_id = i.item_id
scalardb(# ) AS remaining_info GROUP BY name, remaining, items;
     name      | remaining |           items
---------------+-----------+----------------------------
 Suzuki Ichiro |      1500 | {Grape,Orange,Apple,Melon}
 Yamada Hanako |       500 | {Orange,Grape,Mango}
 Yamada Taro   |         0 | {Orange,Melon,Apple}
(3 rows)
```

I think it is better than my sample! (Because my sample does not use any aggregate functions and `GROUP BY` despite it being an OLAP example...)
So, at this time, I adopt your suggestion. Thank you!

@choplin  
As I mentioned above, I applied a new SQL query. At this time, I use it in this document. But, we can update it whenever in the future. So, please feel free to ask me to update it when you find better SQL queries and sample workload!

Reply:
Sure! Anyway, it looks good to me!